src/components/molecules/TableHeader/components/EditTableTitleModal/EditTableTitleModal.tsx

```diff
@@ -0,0 +1,170 @@
+import React, { useCallback, useEffect, useMemo } from "react";
+import { Modal } from "react-bootstrap";
+import { useForm } from "react-hook-form";
+import { useAsyncFn } from "react-use";
+import firebase from "firebase/app";
+
+import { MAX_TABLE_CAPACITY, MIN_TABLE_CAPACITY } from "settings";
+
+import { Table } from "types/Table";
+
+import { useVenueId } from "hooks/useVenueId";
+import { useUser } from "hooks/useUser";
+
+import { InputField } from "components/atoms/InputField";
+
+import "./EditTableTitleModal.scss";
+
+export interface EditTableForm {
+  title: string;
+  subtitle?: string;
+  capacity: number;
+}
+
+export interface EditTableTitleModalProps {
+  isShown: boolean;
+  title: string;
+  subtitle?: string;
+  capacity: number;
+  venueName: string;
+  tables: Table[];
+  tableOfUser?: Table;
+  onHide: () => void;
+}
+
+export const EditTableTitleModal: React.FC<EditTableTitleModalProps> = ({
+  isShown,
+  venueName,
+  title,
+  tables,
+  tableOfUser,
+  subtitle,
+  capacity,
+  onHide,
+}) => {
+  const { user } = useUser();
+  const venueId = useVenueId();
+
+  const formDefaultValues = useMemo(
+    () => ({
+      title,
+      subtitle,
+      capacity,
+    }),
+    [capacity, subtitle, title]
+  );
+
+  const {
+    register,
+    handleSubmit,
+    errors,
+    watch,
+    reset,
+  } = useForm<EditTableForm>({
+    mode: "onBlur",
+    reValidateMode: "onBlur",
+    defaultValues: formDefaultValues,
+  });
+
+  // Updates the default values whenever they change. This is required because they are cached after the first render.
+  useEffect(() => {
+    reset(formDefaultValues);
+  }, [formDefaultValues, reset, title]);
+
+  // use useAsyncFn for easier error handling, instead of state hook
+  const [{ error }, updateTables] = useAsyncFn(
+    async (values: EditTableForm) => {
+      if (!user || !tableOfUser || !venueId) return;
+
+      const updatedTable = {
+        ...tableOfUser,
+        title: values.title,
+        subtitle: values.subtitle,
+        capacity: values.capacity,
+      };
+
+      return await firebase.functions().httpsCallable("venue-updateTables")({
+        venueId,
+        tables,
+        updatedTable,
+      });
+    },
+    [tableOfUser, tables, user, venueId]
+  );
+
+  const values = watch();
+
+  const onSubmit = useCallback(async () => {
+    await updateTables(values).then(() => {
+      onHide();
+    });
+  }, [onHide, updateTables, values]);
+
```

Comment:
We only seem to use `watch()` / `values` here within `onSubmit`, yet looking at the `useForm` documentation, it seems that `handleSubmit` already handles this usage explicitly, so we don't need to do this:

- https://react-hook-form.com/api/useform/watch
- https://react-hook-form.com/api/useform/handlesubmit


There also seems to be no reason to create a `useCallback`'d abstraction for `onSubmit` when we could just use the `useAsyncFn` `saveUpdatesAndCloseModal` directly from `handleSubmit`:

```suggestion
```

Reply:
> There also seems to be no reason to create a useCallback'd abstraction for onSubmit when we could just use the useAsyncFn saveUpdatesAndCloseModal directly from handleSubmit:

Yes, but it's going to look like this:

```tsx
handleSubmit(updateTables(values).then(onHide))
```

Reply:
Do you still want me to do it?

Reply:
Fixed in: https://github.com/sparkletown/sparkle/commit/24da14326be4d256ec9efa8c3472f64899920adb